runtime/vm/LayoutFFITypeHelpers.hpp

```diff
@@ -75,7 +75,52 @@ class LayoutFFITypeHelpers
 	}
 
 	/**
-	 * @brief Convert argument or return type from the type of ffi_type to J9NativeTypeCode
+	 * @brief Get the length of the specified struct layout string by counting the bracket symbols
+	 * for the struct to compute the length.
+	 *
+	 * @param layout[in] A pointer to a c string describing the types of the struct elements.
+	 * e.g. if a struct layout string is "#2[II]", it returns 6 in length.
+	 *
+	 * @return The length of the struct layout string
+	 */
+	static VMINLINE UDATA
+	getLengthOfStructLayout(char *layout)
+	{
```

Comment:
This doesn't need write access to the string, so the parameter type should be `const char *`.
I suggest the assertions should be removed and placed at the call site(s): this should return 0 for malformed input.
We don't need to track both the length and the pointer to the current character.
```c
	/**
	 * @brief Get the length of the specified struct layout string by counting the bracket symbols
	 * for the struct to compute the length.
	 *
	 * For example, if a struct layout string is "#2[II]", it returns 6.
	 *
	 * @param layout[in] A pointer to a C string describing the types of the struct elements
	 * @return The length of the struct layout string
	 */
	static VMINLINE UDATA
	getLengthOfStructLayout(const char *layout)
	{
		UDATA bracketCounter = 0;

		for (const char *cursor = layout;; ++cursor) {
			switch (*cursor) {
			case '\0':
				/* malformed input */
				return 0;
			case '[':
				bracketCounter += 1;
				break;
			case ']':
				if (0 == bracketCounter) {
					/* malformed input */
					return 0;
				}
				bracketCounter -= 1;
				if (0 == bracketCounter) {
					/* Return as we reach the boundary of the specified struct layout string. */
					return cursor - layout + 1;
				}
				break;
			default:
				/* Do nothing. */
				break;
			}
		}
	}
```